arctic/chunkstore/chunkstore.py

```diff
@@ -0,0 +1,376 @@
+        for i, x in enumerate(self._collection.find(spec, sort=[('start', pymongo.ASCENDING)],)):
+            segments.append(decompress(x['data']))
+
+        data = b''.join(segments)
+
+        dtype = PandasSerializer()._dtype(sym['dtype'], sym.get('dtype_metadata', {}))
+        records = np.fromstring(data, dtype=dtype).reshape(sym.get('shape', (-1)))
+
+        data = deserialize(records, sym['type'])
+
+        if chunk_range is None:
+            return data
+        return self.chunker.filter(data, chunk_range)
+
+    def write(self, symbol, item, chunk_size):
+        """
+        Writes data from item to symbol in the database
+
+        Parameters
+        ----------
+        symbol: str
+            the symbol that will be used to reference the written data
+        item: dataframe or series
+            the data to write the database
+        chunk_size: ?
+            A chunk size that is understood by the specified chunker
+        """
+
+        doc = {'_id': bson.ObjectId()}
+        doc['symbol'] = symbol
+        doc['chunk_size'] = chunk_size
+
+        if isinstance(item, Series):
+            doc['type'] = SeriesSerializer.TYPE
+        elif isinstance(item, DataFrame):
+            doc['type'] = DataFrameSerializer.TYPE
+        else:
+            raise Exception("Can only chunk Series and DataFrames")
+
+        records = []
+        ranges = []
+        dtype = None
+
+        for start, end, record in self.chunker.to_chunks(item, chunk_size):
+            r, dtype = serialize(record, string_max_len=self.STRING_MAX)
+            records.append(r)
+            ranges.append((start, end))
+
+        item = np.array([r for record in records for r in record]).flatten()
+        for record in records:
+            if record.dtype.hasobject:
+                raise UnhandledDtypeException()
+
+        sym = self._get_symbol_info(symbol)
+        if sym:
+            # if the symbol already exists, we are basically overwriting it
+            # clean up the data before we orphan the symbol chunks
+            self.delete(symbol)
+
+        doc['dtype'] = str(dtype)
+        doc['shape'] = (-1,) + item.shape[1:]
+        doc['dtype_metadata'] = dict(dtype.metadata or {})
+        doc['len'] = len(item)
+
+        seg_count = 0
+
+        chunks = [r.tostring() for r in records]
+        chunks = compress_array(chunks)
+
+        bulk = self._collection.initialize_unordered_bulk_op()
+        for chunk, rng in zip(chunks, ranges):
+            start = rng[0]
+            end = rng[1]
+            seg_count += 1
+            chunk = {'data': Binary(chunk), 'compressed': True}
+            chunk['start'] = start
+            chunk['end'] = end
+            chunk['symbol'] = symbol
+            chunk['sha'] = checksum(symbol, chunk)
+            bulk.find({'symbol': symbol, 'sha': chunk['sha'], 'start': chunk['start']}
+                      ).upsert().update_one({'$set': chunk, '$addToSet': {'parent': doc['_id']}})
```

Comment:
Looks like this `parent` pointer manipulation is a hang-over from VersionStore.  If you only have one-version do you need this?